core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -101,43 +102,55 @@ trait MethodConfigurationComponent {
       the core methods
      */
 
+    //For readability in tests and whatnot.
+    def create(workspaceContext: SlickWorkspaceContext, newMethodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = upsert(workspaceContext, newMethodConfig)
 
-    //Create is used when importing a new method config or cloning a method config from a different workspace.
-    // It will first check to see if another method config with that namespace and name exist.
-    //    If it does, it will archive the existing and save the given one.
-    //    If it does not, it will simply add the given method config
-    def create(workspaceContext: SlickWorkspaceContext, newMethodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
+    //Looks for an existing method config with the same namespace and name.
+    //If it exists, archives it.
+    //In either case, saves the new method configuration.
+    def upsert(workspaceContext: SlickWorkspaceContext, newMethodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
       uniqueResult[MethodConfigurationRecord](findActiveByName(workspaceContext.workspaceId, newMethodConfig.namespace, newMethodConfig.name)) flatMap {
+        //note that we don't do anything with the version in newMethodConfig. version is a function of location, not the config itself
         case None =>
-          val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, newMethodConfig))
-          configInsert flatMap { configId =>
-            saveMaps(newMethodConfig, configId)
-          }
+          saveWithoutArchive(workspaceContext, newMethodConfig, 1)
         case Some(currentMethodConfigRec) =>
-         save(workspaceContext, currentMethodConfigRec, newMethodConfig)
+          archive(workspaceContext, currentMethodConfigRec) andThen
+            saveWithoutArchive(workspaceContext, newMethodConfig, currentMethodConfigRec.methodConfigVersion + 1)
       }
     } map { _ => newMethodConfig }
 
-    //Update is used when an existing method config is editted, including any renaming of the method config.
-    //The old method config namespace and name are specified because we cannot trust that they are the same as the new method config
-    //First, check if old method config info yields an existing method config
-    //    If it does, archive this existing method config and add the new method config
-    //    If it does not, fail
+    //"Update" the method config at oldMethodConfig[name|namespace] to be the MC - including potential new location - in newMethodConfig.
```

Comment:
OK I guess what I want is a general discussion of the "location" concept regarding MCs.  Do you mean the Workspace it lives in?